spec/lib/fastly_nsq/listener_spec.rb

```diff
@@ -171,7 +171,7 @@ def self.clear
         listener.terminate
 
         state = listener.instance_variable_get(:@done)
-        expect(logger).to have_received(:info).twice
+        expect(logger).to have_received(:info).thrice
```

Comment:
Can we add an expectation

```
expect(consumer).to have_recived(:terminate)
```

or set the exception on listener to have received `cleanup`?

Reply:
good call.
made the tests work but did not make the tests test the new functionality... shame :bell: on me